### PR TITLE
Formatted obj_character

### PR DESCRIPTION
**Pull request by Regulator13**
## Number of Issue Fixed
- #72 partially

### Objects completed
- obj_character

__commit b9778fb__
- obj_localPlayer
- srv_basic
- par_block
- obj_wall
- obj_blockBig
- par_enemy
- obj_patrol
- obj_pirahna

__commit aaf790e__
- par_physics
- obj_ball
- obj_corpse
- obj_block
- obj_jetpack
- obj_water_drop
- obj_score
- obj_rechargeStation
- obj_blockStation
- obj_laser

__commit e0d0eca__
- obj_control
- pos_character
- obj_team
- obj_camera
- obj_tutorial

### Changes Proposed in the Pull Request:
- Changed `camelCased` variables to `underscore_variables`
- Separated blocks of code
- Fixed brace location
- Removed `global.cooperativeMode` artifact
- Changed event descriptions
- Added `///` block descriptions at the beginning of code blocks
- Removed unnecessary comments
- Added comments where ambiguous
- Added `//REMOVE?` and `//TODO` comments for further cleanup and action
- Combined identical `if` statements
- Placed unused code commments in `/* */` blocks
- Blocked together initializations in the `create` event

### Notes for Reviewer
- Check spelling errors in variable names
- Check combinations of `if` statements to make sure they share an identical statement

### CODE REVIEW

File: objects/obj_ball/Collision_968f2037-4728-42b0-993d-16f0a0f697b3.gml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /// @description Attack
 other.hp -= attack;
-if (instance_exists(source)) {
-    other.moochProof = source.team.team;
+if (instance_exists(Source)) {
+    other.moochProof = Source.team.team;
```
Comment (afrey15) [inline objects/obj_ball/Collision_968f2037-4728-42b0-993d-16f0a0f697b3.gml]: Why are a few of your variables being given a capital letter?

File: objects/obj_ball/Create_0.gml [hunk 2]
```diff
@@ -3,8 +3,8 @@
 //attack - attack to hp (-1 for ice, 1 for fire)
 attack = 0;
 
-//source - who created it
-source = noone;
+//Source - who created it
+Source = noone;
```
Comment (afrey15) [inline objects/obj_ball/Create_0.gml]: Capital letter?

File: objects/obj_character/Create_0.gml [hunk 3]
```diff
@@ -1,112 +1,64 @@
-/// @description  initialize variables
-event_inherited();
+/// @description Initializations
```
Comment (afrey15) [inline objects/obj_character/Create_0.gml]: This description is less clear. What about "set variables."? Can we agree on lower case and no punctuation in all comments?

File: objects/obj_character/Create_0.gml [hunk 4]
```diff
@@ -1,112 +1,64 @@
...
-//stop animation
-image_speed = 0;
+mass = 24;
+jumps_max = 1
+jump_height = 8.5
+energy_fire = 5;  //energy cost per fireball
+
+///Physics
+gravity_max = 10; //maximum speed a player can fall
+gravityI = 0.4; //gravity acceleration
```
Comment (afrey15) [inline objects/obj_character/Create_0.gml]: Camel case. Can this variable be given a second word besides I that better describes it?

File: objects/obj_character/Create_0.gml [hunk 5]
```diff
@@ -1,112 +1,64 @@
...
+frame_step = 0; //frame_step - current frame for animation
+frame_buffer = 0; //frame_buffer - slow animation
+frame_buffer_max = 4;
+dir = 0; //Direction the player is facing, left= -1, right =1
+mooch_buffer_max = 120;
+mooch_buffer = 0;
+
+///Input
+player_input = 0; //number of player input in input array
+Input_player = noone; //player object where to get input from, local or network
```
Comment (afrey15) [inline objects/obj_character/Create_0.gml]: Capital letter?

File: objects/obj_control/Other_4.gml [hunk 6]
```diff
@@ -6,18 +6,10 @@ if (global.gameStart) {
...
-        global.water_rate = .05;
-        }
-    else {
-        //water delay
-        global.waterDelay = 1000;
-        //water rate
-        global.water_rate = .1;
-        }
+	//Set the water delay and rate
+    global.waterDelay = 1000;
```
Comment (afrey15) [inline objects/obj_control/Other_4.gml]: Snake Case has not been corrected

File: objects/obj_pirahna/Step_0.gml [hunk 7]
```diff
@@ -5,57 +5,57 @@
...
-                jumpPressed = true;
+            if(Input_player.inputs[LEFT_KEY]) haxis1 = -1;
+            if(Input_player.inputs[RIGHT_KEY]) haxis1 = 1;
+            if(Input_player.inputs[UP_KEY] == KEY_PRESSED) vaxis1 = -1;
+            if(Input_player.inputs[DOWN_KEY] == KEY_PRESSED) vaxis1 = 1;
+            if(Input_player.inputs[ACTION_KEY] == KEY_PRESSED) {
+                jump_pressed = true;
                 //unpress key
-                InputPlayer.inputs[ACTION_KEY] = KEY_RELEASED;
+                Input_player.inputs[ACTION_KEY] = KEY_RELEASED;
```
Comment (afrey15) [inline objects/obj_pirahna/Step_0.gml]: Capital letter?

File: objects/obj_ball/Collision_968f2037-4728-42b0-993d-16f0a0f697b3.gml (hunk 1 above)
Comment (Regulator13) [reply objects/obj_ball/Collision_968f2037-4728-42b0-993d-16f0a0f697b3.gml]: This variable holds an instance id (as shown by the `instance_exists()` function. It should be capitalized.

File: objects/obj_ball/Create_0.gml (hunk 2 above)
Comment (Regulator13) [reply objects/obj_ball/Create_0.gml]: Instance id

File: objects/obj_character/Create_0.gml (hunk 3 above)
Comment (Regulator13) [reply objects/obj_character/Create_0.gml]: Set variables is clearer. This however is a description so I prefer it be capitalized. I'd say anything that is `///` could be capitalized and maybe lowercase for `//`. Although, they are comments so we probably shouldn't be too nit-picky on their formatting

File: objects/obj_character/Create_0.gml (hunk 4 above)
Comment (Regulator13) [reply objects/obj_character/Create_0.gml]: This is camelCased but also its a single character which is why I didn't change it. I guess `gravity_i` isn't terrible but `gravity_incr` might be better

File: objects/obj_character/Create_0.gml (hunk 5 above)
Comment (Regulator13) [reply objects/obj_character/Create_0.gml]: Instance id again

File: objects/obj_control/Other_4.gml (hunk 6 above)
Comment (Regulator13) [reply objects/obj_control/Other_4.gml]: good catch, I'll fix that

File: objects/obj_pirahna/Step_0.gml (hunk 7 above)
Comment (Regulator13) [reply objects/obj_pirahna/Step_0.gml]: more instance id

File: objects/obj_blockBig/Step_0.gml [hunk 8]
```diff
@@ -1,16 +1,14 @@
-/// @description  @description Hp
+/// @description  @description Blow Up
 
 //Blow up block
 if (hp <= 0){
     hp = 0;
```
Comment (Phabbits) [inline objects/obj_blockBig/Step_0.gml]: Can we get a comment for this statement? I am pretty sure it is supposed to help with animation.

Comment (Regulator13) [reply objects/obj_blockBig/Step_0.gml]: If you could make a suggestion that'd be good

**Comment by Regulator13** [comment]
Hey @Phabbits, I'm trying to add in comments as I go but I also don't want to be misleading, as I don't understand everything. Also remember if the variable has a self-explanatory name, no comment is needed to define it, `gravity_max` for example.

**Comment by afrey15** [comment]
I suggest we merge before this pull request gets out of hand. It is annoying to scroll through hundreds of lines we've already reviewed in order to review new additions and there is no way to filter the new commits during a review.

**Comment by Regulator13** [comment]
We can merge it if you approve the changes. Also you should know there _is_ a way to filter commits, you can just go to the commits tab and click on the latest one and it will only so you the differential for that commit. Also the commits are linked in the description of changes at the top above the resources they changed.

**Comment by afrey15** [comment]
Will merge. However, if you try doing a review while filtering any commits, it will not let you start a review

**Pull request merged**